src/cpp/FilesInformation/Loaders/FilesByNameLoader.cpp

```diff
@@ -34,67 +34,92 @@ bool FilesByNameLoader::isTextFile(const std::string &filepath) {
     FilesByNameLoader loader = FilesByNameLoader("\\");
-    File file = loader.loadFile("C:\\a\\b\\c\\file.txt");
-    CHECK(file.filename() == "file");
-    CHECK(file.extension() == "txt");
+    File *file = loader.loadFile("C:\\a\\b\\c\\file.txt");
+    CHECK(file->filename() == "file");
+    CHECK(file->extension() == "txt");
+    delete file;
 }
 
 TEST_CASE("Just file name") {
     FilesByNameLoader loader = FilesByNameLoader();
-    File file = loader.loadFile("file.txt");
-    CHECK(file.filename() == "file");
-    CHECK(file.extension() == "txt");
+    File *file = loader.loadFile("file.txt");
+    CHECK(file->filename() == "file");
+    CHECK(file->extension() == "txt");
+    delete file;
 }
 
 TEST_CASE("Extension with several dots") {
     FilesByNameLoader loader = FilesByNameLoader();
-    File file = loader.loadFile("C:/sd/fd/f/file.txt.txt");
-    CHECK(file.filename() == "file");
-    CHECK(file.extension() == "txt.txt");
+    File *file = loader.loadFile("C:/sd/fd/f/file.txt.txt");
+    CHECK(file->filename() == "file");
+    CHECK(file->extension() == "txt.txt");
+    delete file;
 }
 
 TEST_CASE("Extension with several dots with Windows separator") {
     FilesByNameLoader loader = FilesByNameLoader("\\");
-    File file = loader.loadFile("C:\\sd\\fd\\f\\file.txt.txt");
-    CHECK(file.filename() == "file");
-    CHECK(file.extension() == "txt.txt");
+    File *file = loader.loadFile("C:\\sd\\fd\\f\\file.txt.txt");
+    CHECK(file->filename() == "file");
+    CHECK(file->extension() == "txt.txt");
+    delete file;
 }
 
 TEST_CASE("Existing file") {
     FilesByNameLoader loader = FilesByNameLoader();
-    File file = loader.loadFile("resources/text.txt");
-    CHECK(file.exists());
+    File *file = loader.loadFile("resources/text.txt");
+    CHECK(file->exists());
+    delete file;
 }
 
 TEST_CASE("Existing text file is text check") {
     FilesByNameLoader loader = FilesByNameLoader();
-    File file = loader.loadFile("resources/text.txt");
-    CHECK(file.is_text_file());
+    File *file = loader.loadFile("resources/text.txt");
+    CHECK(file->is_text_file());
+    delete file;
 }
 
 TEST_CASE("Existing binary file is text check") {
     FilesByNameLoader loader = FilesByNameLoader();
-    File file = loader.loadFile("resources/binary.bin");
-    CHECK(!file.is_text_file());
+    File *file = loader.loadFile("resources/binary.bin");
+    CHECK(!file->is_text_file());
+    delete file;
 }
 
 TEST_CASE("Non Existing file") {
     FilesByNameLoader loader = FilesByNameLoader();
-    File file = loader.loadFile("resources/text.txt.txt.txt");
-    CHECK(!file.exists());
-}
\ No newline at end of file
+    File *file = loader.loadFile("resources/text.txt.txt.txt");
+    CHECK(!file->exists());
+    delete file;
+}
+
+
+TEST_CASE("Empty directory") {
+    FilesByNameLoader loader;
+    Directory* directory = loader.loadDirectory("resources/empty_directory_test");
+    CHECK(directory->size() == 0);
```

Comment:
Currently this test fails. It is impossible to commit empty directory to git repository - so as you can check e.g. from GitHub web interface, there is no resources/empty_directory_test in repository.

Perhaps better approach for testing would be: 

1. Create a new directory
2. Check that everything works as it should
3. Remove the created directory
(The same approach could be used for all tests related to filesystem - it may seem strange to put some special files into repository just for testing. But for this particular test it seems to be required)

Reply:
Also current behavior seems to be: when calling FilesByNameLoader::loadDirectory() on non-existing path - it throws an exception. If this is an intended behavior - perhaps another test is needed, with CHECK_THROWS_AS assertion to show the type of exception